virtcontainers/factory/factory.go

```diff
@@ -67,6 +68,16 @@ func NewFactory(config Config, fetchOnly bool) (vc.Factory, error) {
 	return &factory{b}, nil
 }
 
+// SetLogger sets the logger for the factory.
+func SetLogger(logger logrus.FieldLogger) {
+	fields := logrus.Fields{
+		"source": "virtcontainers",
+		"arch":   runtime.GOARCH,
```

Comment:
I see this is basically doing the same as `virtcontainers/api.go:SetLogger()`. But I'm wondering why we don't just add the `arch` field here:

- https://github.com/kata-containers/runtime/blob/master/cli/main.go#L146

But also, you might want to add a `"subsystem": "factory"` field as we've done that in various parts of vc (like `network.go`).

Reply:
Good point about moving arch to main.go.

As for `subsystem`, it is set in `factory.log()` function.

Reply:
Updated. PTAL @jodh-intel .